### PR TITLE
feat: Upgrade tooling to support TypeScript

### PR DESCRIPTION
**Pull request by ibgreen**
For #5589 

#### Background
- Update various tooling scripts and configs
#### Change List
- Update various tooling scripts and configs

Note: the big eslint and webpack configs are copied from loaders.gl, they should really be pushed back into the ocular-dev-tools defaults once things are working

### CODE REVIEW

File: scripts/bundle.config.js [hunk 1]
```diff
@@ -109,7 +113,7 @@ module.exports = (env = {}) => {
   }
 
   // NOTE uncomment to display config
-  // console.log('webpack config', JSON.stringify(config, null, 2));
+  console.log('webpack config', JSON.stringify(config, null, 2));
```
Comment (Pessimistress) [inline scripts/bundle.config.js:116]: Comment out?

File: test/modules/aggregation-layers/contour-layer/marching-squares.spec.js [hunk 2]
```diff
@@ -407,13 +407,13 @@ test('MarchingSquares#getCode', t => {
       width: testCase.gridSize ? testCase.gridSize[0] : gridSize[0],
       height: testCase.gridSize ? testCase.gridSize[1] : gridSize[1]
     });
-    t.equals(code, testCase.code, `Code: expected: ${testCase.code}, actual: ${code}`);
+    t.equals(code, testCase.code, `Code: expected=${testCase.code}, actual=${code}`);
```
Comment (Pessimistress) [inline test/modules/aggregation-layers/contour-layer/marching-squares.spec.js:410]: Just curious, why are these a problem?

File: .ocularrc.js [hunk 3]
```diff
@@ -17,8 +17,7 @@ const useLocalLuma = false;
 
 const config = {
   lint: {
-    paths: ['modules', 'examples', 'test'],
-    extensions: ['js']
+    paths: ['modules'] // , 'test', 'examples',
```
Comment (Pessimistress) [inline .ocularrc.js:20]: Can we at least include `test`?

File: test/modules/aggregation-layers/contour-layer/marching-squares.spec.js (hunk 2 above)
Comment (ibgreen) [reply test/modules/aggregation-layers/contour-layer/marching-squares.spec.js:410]: > Just curious, why are these a problem?

Non-technical reason: I search for the tape generated string "expected: " to find errors in the long test logs, so prefer if the tests don't print "expected: " as it slows down my iterations...

File: .ocularrc.js (hunk 3 above)
Comment (ibgreen) [reply .ocularrc.js]: > Can we at least include test?

done

**Review by Pessimistress — APPROVED** [review]

**Comment by ibgreen** [comment]
Will remove the `time` calls in a new PR.

**Pull request merged**